projects/raytracing/030_raytracing_refract_vulkan/030_raytracing_refract_vulkan.cpp

```diff
@@ -0,0 +1,1551 @@
+#include "window.h"
+
+#include "vk_renderer.h"
+#include "tri_mesh.h"
+
+#include <glm/glm.hpp>
+#include <glm/matrix.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+#include <glm/gtx/transform.hpp>
+using namespace glm;
+
+#define CHECK_CALL(FN)                               \
+    {                                                \
+        HRESULT hr = FN;                             \
+        if (FAILED(hr)) {                            \
+            std::stringstream ss;                    \
+            ss << "\n";                              \
+            ss << "*** FUNCTION CALL FAILED *** \n"; \
+            ss << "FUNCTION: " << #FN << "\n";       \
+            ss << "\n";                              \
+            GREX_LOG_ERROR(ss.str().c_str());        \
+            assert(false);                           \
+        }                                            \
+    }
+
+// =============================================================================
+// Macros, enums, and constants
+// =============================================================================
+const uint32_t kOutputResourcesOffset = 0;
+const uint32_t kGeoBuffersOffset      = 20;
+const uint32_t kIBLTextureOffset      = 3;
+
+// =============================================================================
+// Globals
+// =============================================================================
+static uint32_t gWindowWidth  = 1920;
```

Comment:
Lets drop these down to 1280x720. 
1920 is a bit too big when using for presentations.